src/h3/mod.rs

```diff
@@ -1309,10 +1411,69 @@ mod tests {
 
             self.pipe.advance(buf).ok();
 
+            // server streams
+            self.server.send_settings(&mut self.pipe.server)?;
+            self.pipe.advance(buf).ok();
+
+            self.server
+                .open_qpack_encoder_stream(&mut self.pipe.server)?;
+            self.pipe.advance(buf).ok();
+
+            self.server
+                .open_qpack_decoder_stream(&mut self.pipe.server)?;
+            self.pipe.advance(buf).ok();
+
+            if self.pipe.server.grease {
+                self.server.open_grease_stream(&mut self.pipe.server)?;
+            }
+
+            self.pipe.advance(buf).ok();
+
+            loop {
+                match self.client.poll(&mut self.pipe.client) {
```

Comment:
I think you could just turn this into something like:
```rust
while self.client.poll(&mut self.pipe.client).is_ok() {
  // Do nothing.
}
```

Same for the one below.